libraries/chain/fork_database.cpp

```diff
@@ -703,7 +655,8 @@ namespace eosio::chain {
    fork_database::fork_database(const std::filesystem::path& data_dir)
       : data_dir(data_dir)
         // genesis starts with legacy
-      , fork_db_l{std::make_unique<fork_database_legacy_t>(fork_database_legacy_t::legacy_magic_number)}
+      , fork_db_l{std::make_unique<fork_database_legacy_t>()}
+      , fork_db_s{std::make_unique<fork_database_if_t>()}
```

Comment:
I'm surprised to see we always create both. After savanna is activated we will not need legacy. Would be nice to not create either unless needed. Looks like you are creating them on open already.

Reply:
I don't think it matters if we create them empty, they are very small. Actually probably there is no point for them to be unique_ptr, we could store the two values directly.